wordpress-mobile/WordPress-Android
# Move photon url logic from VH to VM

Fixes #9639 

This PR shouldn't change the behavior. It just moves `getResizedImageUrl` from ViewHolder into ViewModel.
 - removes unnecessary postViewHolderConfig
- removes unnecessary reference to `siteModel` from PostListFragment
- introduces injectable wrapper around ReaderUtils


To test:
Make sure featured images on post list items are being loaded

Update release notes:

- No user facing changes
